### PR TITLE
DPage.inc: Use modern DPDatabase::query, escaping, sprintf

### PR DESCRIPTION
**Pull request by bpfoley**
This was tricky. I've treble checked the diffs and tested various parts of the proofing interface, and found some mistakes along the way. Extra eyeballs appreciated though.

### CODE REVIEW

File: pinc/DPage.inc [hunk 1]
```diff
@@ -792,20 +900,26 @@ function _project_adjust_n_available_pages( $projectid, $adjustment )
         }
     }
 
-    mysqli_query(DPDatabase::get_connection(), "
+    $sql = sprintf("
         UPDATE projects
-        SET n_available_pages = $expr
-        WHERE projectid='$projectid'
-    ") or die(DPDatabase::log_error());
+        SET n_available_pages = %s
```
[the remark this answers is not in the document]
Comment (bpfoley) [reply pinc/DPage.inc:905]: Yes! %d

File: pinc/DPage.inc [hunk 2]
```diff
@@ -86,15 +90,19 @@ function Pages_prepForRound( $projectid, $round_number )
         // how project_add_page() sets page-states.
 
         $timestamp = time();
-
-        $result = mysqli_query(DPDatabase::get_connection(), "
-            UPDATE $projectid
+        $sql = sprintf("
+            UPDATE %s
```
Comment (cpeel) [inline pinc/DPage.inc:94]: Embed `$projectid` here in the string directly.

File: pinc/DPage.inc [hunk 3]
```diff
@@ -86,15 +90,19 @@ function Pages_prepForRound( $projectid, $round_number )
...
             SET
-                state = '{$round->page_avail_state}',
-                {$round->time_column_name} = $timestamp
-        ");
-    
-        _project_adjust_n_available_pages( $projectid, 'ALL' );
+                state = '%s',
+                %s = %d",
```
Comment (cpeel) [inline pinc/DPage.inc:97]: Embed the column name directly, per https://www.pgdp.net/w/index.php?title=DP_Code_Best_Practices#Column_names_as_variables

File: pinc/DPage.inc (hunk 2 above)
Comment (cpeel) [reply pinc/DPage.inc:94]: Here and elsewhere.

File: pinc/DPage.inc (hunk 3 above)
Comment (cpeel) [reply pinc/DPage.inc:97]: Here and elsewhere.

Comment (bpfoley) [reply pinc/DPage.inc]: I ended up with some places where I had to build a bunch of temporary strings and concatenate them together to satisfy this, so I replaced them with a a couple of smaller helper functions. Let me know if you dislike the `$setters` code.

**Review by srjfoo — APPROVED** [review]
I've worked through testing, and everything I was able to figure out how to test seems to work now.

File: pinc/DPage.inc [hunk 4]
```diff
@@ -434,14 +450,15 @@ function Page_markAsBad( $projectid, $image, $round, $user, $reason )
...
-    _Page_UPDATE( $projectid, $image, "
-        state='{$round->page_bad_state}',
-        b_user='$user',
-        b_code=$reason,
-        {$round->text_column_name}={$round->prevtext_column_name}
-    " );
+    $setters = join(", ", [
+        set_col_str("state", $round->page_bad_state),
+        set_col_str("b_user", $user),
+        set_col_str("b_code", $reason),
```
Comment (cpeel) [inline pinc/DPage.inc:456]: `set_col_num()`

File: pinc/DPage.inc [hunk 5]
```diff
@@ -453,16 +470,17 @@ function Page_eraseBadMark( $projectid, $image, $round, $user )
...
-    _Page_UPDATE( $projectid, $image, "
-        state='{$round->page_avail_state}',
-        b_user='',
-        b_code=0,
-        {$round->user_column_name}=''
-    " );
+    $setters = join(", ", [
+        set_col_str("state", $round->page_avail_state),
+        set_col_str("b_user", ""),
+        set_col_str("b_code", 0),
```
Comment (cpeel) [inline pinc/DPage.inc:476]: `set_col_num()`

File: pinc/DPage.inc [hunk 6]
```diff
@@ -86,15 +90,15 @@ function Pages_prepForRound( $projectid, $round_number )
         // how project_add_page() sets page-states.
 
         $timestamp = time();
-
-        $result = mysqli_query(DPDatabase::get_connection(), "
+        $sql = "
```
Comment (cpeel) [inline pinc/DPage.inc:93]: Is there a reason not to use `sprintf()` here?

**Review by cpeel — COMMENTED** [review]
I really like the setter helper functions!

File: pinc/DPage.inc (hunk 6 above)
Comment (bpfoley) [reply pinc/DPage.inc:93]: Not really except it was interleaved with column names. But hey, I can use setters here again!

**Pull request merged**